### PR TITLE
Cookie vignette

### PR DESCRIPTION
**Pull request by rmbielby**
Adding a vignette / article page to the pkgdown site explaining how to apply the cookie consent to a dashboard:

Main changes:

- Description of how to implement cookie consent
- Removed outdated description from readme

Incidental change: 

- Added explicit setting of an expiry on the cookie consent cookie (365 days)

### CODE REVIEW

File: vignettes/implementing-cookies.Rmd [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Cookie consent is a mandatory requirement for any dashboard when using cookies
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:17]: This sentence is a bit clunky I think? Also I know most of us know what cookies are but is it worth a one sentence explanation of what they are/do?

File: vignettes/implementing-cookies.Rmd [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+)
+```
+
+Cookie consent is a mandatory requirement for any dashboard when using cookies 
+and typically all DfE dashboards will be using cookies as part of tracking with
+Google Analytics and so will require cookie consent.
+
+> Note that to activate Google Analytics on your data dashboard, you will need 
+to contact the Statistics Development team at 
+[statistics.development@education.gov.uk](mailto:statistics.development@education.gov.uk) and request a Google Analytics key.
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:23]: Is this still us or is it an EES team job now?

File: vignettes/implementing-cookies.Rmd [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+dfeshiny provides a standard GDS compliant banner, a cookie information panel
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:25]: Nitpicky but recommend writing GDS out in full the first time we use it

File: vignettes/implementing-cookies.Rmd [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+Google Analytics and so will require cookie consent.
+
+> Note that to activate Google Analytics on your data dashboard, you will need 
+to contact the Statistics Development team at 
+[statistics.development@education.gov.uk](mailto:statistics.development@education.gov.uk) and request a Google Analytics key.
+
+dfeshiny provides a standard GDS compliant banner, a cookie information panel 
+and the underlying code required to allow end users to give or withhold their 
+consent for cookies to be used. If they withhold their consent, then tracking 
+with Google Analytics will be halted.
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:28]: What are the implications of this for analysts? Do they need to know?

File: vignettes/implementing-cookies.Rmd [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+including in your dashboard's www/ folder, which can be achieved by running the 
+following command from the R console in your dashboard project / repo:
+
+```{r eval=FALSE}
+dfeshiny::init_cookies()
+```
+
+Before adding the cookie consent banner, we recommend adding your Google 
+Analytics key as a variable to your dashboard's **[global.R](https://github.com/dfe-analytical-services/dfeshiny/blob/cookie-module/tests/test_dashboard/global.R)**
+file (replacing `ABCDE12345` below with the key provided by the Statistics
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:44]: Again not sure if this needs to be EES team 😄

File: vignettes/implementing-cookies.Rmd [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+Before adding the cookie consent banner, we recommend adding your Google 
+Analytics key as a variable to your dashboard's **[global.R](https://github.com/dfe-analytical-services/dfeshiny/blob/cookie-module/tests/test_dashboard/global.R)**
+file (replacing `ABCDE12345` below with the key provided by the Statistics 
+Development team):
+
+```{r eval=FALSE}
+google_analytics_key <- "ABCDE12345"
+```
+
+You will then need to add the necessary code to your **ui.R** file to a) run the
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:51]: Worth a reminder to the user to edit the "Your dashboard name" bit? Maybe seems overkill but also I have seen people overlook stuff like that!

File: vignettes/implementing-cookies.Rmd [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+In the above you should keep all of the flags as they are shown. Note that this 
+assumes you have added `google_analytics_key` as a declared variable in your 
+global.R file. If you've not done this, you'll need to replace 
+`google_analytics_key = google_analytics_key` to 
+`google_analytics_key = "ABCDE12345"` (replacing "ABCDE12345" with your own 
+key).
+
+With the above included, when the user's preference will itself be stored in a
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:106]: I think the "when" in this sentence could go!

**Review by jen-machin — COMMENTED** [review]
Apologies for nitpicky comments, hopefully some fairly minor changes here

File: vignettes/implementing-cookies.Rmd (hunk 2 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:23]: EES team - updated

File: vignettes/implementing-cookies.Rmd (hunk 3 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:25]: Not nitpicky, good point. Updated.

File: vignettes/implementing-cookies.Rmd (hunk 4 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:28]: Added info in brackets.

File: vignettes/implementing-cookies.Rmd (hunk 5 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:44]: Reckon so. Now updated.

File: vignettes/implementing-cookies.Rmd (hunk 6 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:51]: Fair, added.

File: vignettes/implementing-cookies.Rmd (hunk 7 above)
Comment (rmbielby) [reply vignettes/implementing-cookies.Rmd:106]: Removed.

File: vignettes/implementing-cookies.Rmd [hunk 8]
```diff
@@ -41,16 +48,15 @@ dfeshiny::init_cookies()
 
 Before adding the cookie consent banner, we recommend adding your Google 
 Analytics key as a variable to your dashboard's **[global.R](https://github.com/dfe-analytical-services/dfeshiny/blob/cookie-module/tests/test_dashboard/global.R)**
-file (replacing `ABCDE12345` below with the key provided by the Statistics 
-Development team):
+file (replacing `ABCDE12345` below with the key provided by the Explore Education Statistics Plaforms team):
```
Comment (jen-machin) [inline vignettes/implementing-cookies.Rmd:51]: very minor typo on platforms

**Review by jen-machin — APPROVED** [review]
just that one typo to change but everything else looks great!

**Pull request merged**